rust/src/composepost.rs

```diff
@@ -128,11 +128,14 @@ pub(crate) fn compose_postprocess_final(rootfs_dfd: i32) -> CxxResult<()> {
 
 #[context("Handling treefile 'units'")]
 fn compose_postprocess_units(rootfs_dfd: &openat::Dir, treefile: &mut Treefile) -> Result<()> {
-    let multiuser_wants = Path::new("usr/etc/systemd/system/multi-user.target.wants");
+    let multiuser_wants = Path::new("etc/systemd/system/multi-user.target.wants");
 
     let mut created = false;
     for unit in treefile.parsed.units.iter().flatten() {
         if !created {
+            if !rootfs_dfd.exists("etc")? {
```

Comment:
I guess this whole "check for /etc and then create subdir" logic would be better moved out of the for-loop.

Reply:
The idea is we don't touch `/etc` unless we're explicitly asked to.

Reply:
IOW if you don't use the `units:` directive in a treefile, I don't think we should be creating `/etc/systemd/system/multi-user.target.wants`.

Reply:
Fair point.